core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -2015,27 +2015,38 @@ private[spark] object Utils
     }
   }
 
+  private[util] val threadInfoOrdering = Ordering.fromLessThan {
+    (threadTrace1: ThreadInfo, threadTrace2: ThreadInfo) => {
+      def priority(ti: ThreadInfo): Int = ti.getThreadName match {
+        case name if name.contains("Executor task launch") => 3
+        case "Driver" => 2
```

Comment:
https://github.com/apache/spark/pull/46962#issuecomment-2163311959

not related to this pr: Although there is a test case, it might be better to make the literal of this identifier linkable.

Reply:
sounds reasonable, thanks for suggestion